internal/tsetse/rules/equals_nan_rule.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @fileoverview Bans `== NaN` in TypeScript code, since it is always false for
+ * any value.
+ */
+
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import {Checker} from '../checker';
+import {AbstractRule} from '../rule';
+
+const FAILURE_STRING = 'x == NaN always returns false; use isNaN(x) instead';
+
+export class Rule extends AbstractRule {
+  readonly ruleName = 'equals-nan';
+  readonly code = 21222;
+
+  register(checker: Checker) {
+    checker.on(
+        ts.SyntaxKind.BinaryExpression, checkBinaryExpression, this.code);
+  }
+}
+
+function checkBinaryExpression(checker: Checker, node: ts.BinaryExpression) {
+  if ((node.operatorToken.kind === ts.SyntaxKind.EqualsEqualsToken ||
+       node.operatorToken.kind === ts.SyntaxKind.EqualsEqualsEqualsToken) &&
```

Comment:
update comments/docs to say it covers triple-equals also

Reply:
Updated the `@fileoverview` and failure string. Do you think it should also be mentioned in the failure string? It seems a little verbose.

Reply:
yeah, I think longer is worth it to be more exact